src/store/useStore.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable sort-keys */
+import { create } from 'zustand';
+import { api } from '@/hooks';
+import { TSortByValues } from '@/utils';
+
+type State = {
+    filteredCurrency: string;
+    hasCreatedAdvertiser: boolean;
+    selectedPaymentMethods: string[];
+    shouldUseClientLimits: boolean;
+    sortByValue: TSortByValues;
+    userInfoState:
+        | {
+              error: ReturnType<typeof api.advertiser.useGetInfo>['error'];
+              isActive: boolean;
+              isIdle: boolean;
+              isLoading: boolean;
+          }
+        | undefined;
+};
+
+type Action = {
+    setFilteredCurrency: (filteredCurrency: string) => void;
+    setHasCreatedAdvertiser: (hasCreatedAdvertiser: boolean) => void;
+    setSelectedPaymentMethods: (selectedPaymentMethods: string[]) => void;
+    setShouldUseClientLimits: (shouldUseClientLimits: boolean) => void;
+    setSortByValue: (sortByValue: TSortByValues) => void;
+    setUserInfoState: (userInfoState: State['userInfoState']) => void;
+};
+
+const useStore = create<Action & State>()(set => ({
+    filteredCurrency: '',
+    hasCreatedAdvertiser: false,
+    selectedPaymentMethods: [],
+    shouldUseClientLimits: true,
+    sortByValue: 'rate',
+    userInfoState: undefined,
+    setFilteredCurrency: (filteredCurrency: string) => set({ filteredCurrency }),
+    setHasCreatedAdvertiser: (hasCreatedAdvertiser: boolean) => set({ hasCreatedAdvertiser }),
+    setSelectedPaymentMethods: (selectedPaymentMethods: string[]) => set({ selectedPaymentMethods }),
+    setShouldUseClientLimits: (shouldUseClientLimits: boolean) => set({ shouldUseClientLimits }),
+    setSortByValue: (sortByValue: TSortByValues) => set({ sortByValue }),
+    setUserInfoState: userInfoState => set({ userInfoState }),
+}));
```

Comment:
How I chose to separate the setters and their values, was to separate the types into two categories. States and Actions. Hence why I chose to add the ignore for sorting these values above.

However if the store does get way too big, another great thing Zustand provides is the ability to create "slices". Similar to how we had multiple stores before in v1, this concept is more towards grouping the State and Actions together in separate files, and then combining them all in the store! This way we can group certain State and Actions to certain behaviours within our App like Buy/Sell Filters, AdvertiserInfoState, etc

You can read more about it here: https://docs.pmnd.rs/zustand/guides/slices-pattern